### DISCUSSION TITLE
Position neu angelegter Strukturelemente

### DISCUSSION THREAD

**Discussion by andre-hohmann** [kitodo/kitodo-production#5675] [Q&A]
**Beschreibung**

In Kitodo.Production 3 bestehen mehrere Möglichkeiten, um Strukturelemente anzulegen. 
1. Auswahl mehrerer Bildobjekte in der Galerie und Erstellen des Strukturelements (Siehe Beispiel 1)
2. Auswahl mehrerer Bildobjekte in der Scrollbar der Detailansicht und Erstellen des Strukturelements
3. Erstellen eines Strukturelements im Strukturbaum (Siehe Beispiel 2)

In den Fällen 1 und 2 werden die Strukturelemente jeweils an der Position des ersten enthaltenen Bildobjekts erstellt. 
In dem Fall 3 können die Strukturelemente nur am Anfang oder am Ende der Bildobjekte erstellt werden. Sind einzelne Bildobjekte zwischen den Strukturelementen vorhanden, müssen die Strukturelemente an die jeweils korrekte Stelle geschoben werden. Ansonsten werden die Bildobjekte nicht in der korrekten Reihenfolge angezeigt. 

**Erwartung**

Auch im Fall 3 soll es möglich sein, dass die Strukturelemente an der Position des ersten enthaltenen Bildobjekts erstellt werden. Entweder, indem dies automatisch durchgeführt wird, oder indem in der Auswahlliste "Position" neben "Als letztes Subelement ...", "Als erstes Subelement ...", ... auch die Option "Position des ersten enthaltenen Objekts" hinzugefügt wird. 

**Fragen**

1. Ist es gewollt, dass im Strukturbaum das Strukturelement nicht an der Position des ersten Bildobjekts erstellt wird?
2. Besteht auch in anderen Einrichtungen Bedarf an der Anpassung der Position von neu angelegten Strukturelementen?

**Beispiele**

Beispiel 1
![StrukturelementNeuGalerie](https://github.com/kitodo/kitodo-production/assets/14835770/4aa6360c-a2e2-47b9-b253-31696df57bb6)

Beispiel 2
![StrukturelementNeuStrukturbaum](https://github.com/kitodo/kitodo-production/assets/14835770/5542fed3-4f5d-44e7-a46c-98057d0eca0f)

**Comment by apiller** [comment]
Könnte es ein, dass es bei dem zweiten Beispiel daran liegt, dass die anderen Bilder "unstrukturiert" sind? Die unstrukturierten Medien werden immer unten angezeigt. Ich hab es mal kurz mit komplett durchstrukturierten Ausgaben getestet da landet das Strukturemelent wo es hingehört.

**Comment by andre-hohmann** [comment]
Ja, das ist ein Faktor. Bei den unstrukturierten Medien muss jedoch berücksichtigt werden, ob sie im Strukturbaum oder in der Galerie "unten" angezeigt werden. 

1. In der Galerie ist es in Ordnung, weil es nur die Darstellung im Metadateneditor betrifft. 
2. Im Strukturbaum führt es zu Problemen, weil die Sortierung beim Export übernommen wird und die unstrukturierten Medien auch in der Präsentation entsprechend angezeigt werden. 

Bezüglich 2 haben wir die Migration-Transformation angepasst, so dass die unstrukturierten Medien und die Strukturelemente jeweils an der korrekten Position sortiert werden. Wir wollen unbedingt die korrekte Reihenfolge der Medien anzeigen. 
- #5518

Wenn zum Beispiel folgende Zuordnung realisiert werden soll, muss man aus meiner Sicht manuell nacharbeiten, weil Strukturelemente immer aufeinanderfolgen, wenn sie im Strukturbaum erstellt werden.  
- Objekt 1: Einband
- Objekt 2: unstrukturiert
- Objekt 3: unstrukturiert
- Objekt 4: Titelblatt
- Objekt 5: unstrukturiert
- Objekt 6: unstrukturiert
- Objekt 7: unstrukturiert
- Objekt 8: Inhaltsverzeichnis
- Objekt 9: unstrukturiert
- Objekt 10: unstrukturiert
- Objekt 11: unstrukturiert 
- Objekt 12: Kapitel 1
- Objekt 13: Kapitel 2
- ...

**Comment by solth** [comment]
Das unter Beispiel 2 beschriebene Phänomen hängt mit #3629 zusammen. Das (funktionierende) Erstellen von Strukturelementen aus (mehreren) ausgewählten Seiten in der Galerie ist davon abhängig, dass es sich um aufeinanderfolgende Seiten im selben Strukturelement handelt. Hier wird das Strukturelement durch Rechtsklick auf die selektierten Seiten in der _Galerie_ angelegt. 

Beim _Strukturbaum_ existiert diese Option auch (siehe Screenshot), da hier aber immer nur _eine_ Seite gleichzeitig ausgewählt werden kann, macht die Option (noch) nicht soviel Sinn wie in der Galerie. (Alternativ ist es möglich, mehrere Seiten in der Galerie auszuwählen, die Option, daraus ein Strukturelement an der gleichen Stelle zu erzeugen, dann aber über die - eine - im Strukturbaum markierte Seite zu erstellen. Das wird aber den hier geschilderten Anforderungen vermutlich nicht gerecht, da es ja darum geht, diese Aktionen komplett - also inkl. Selektion - über den Strukturbaum ausführen zu können).

Beim in Beispiel 1 beschriebenen Vorgehen dagegen wird ja das _Elternelement_ (hier die Monographie) ausgewählt und für diese aus bestimmten Seiten ein Kindstrukturelement erzeugt. Da wir bei der Selektion im Baum aber noch nicht mehrere Seiten gleichzeitig auswählen können (siehe oben erwähntes Ticket #3629) können wir in der Folge auch nicht sicherstellen, dass _aufeinanderfolgende_ Seiten ohne z.B. ein Kindelement dazwischen ausgewählt wurden und dadurch die Option "Strukturelement an aktueller Position erstellen" angeboten werden soll. Das Problem wird sich also zusammen mit der Behebung von #3629 hoffentlich lösen lassen.

_Kontextmenü-Option im Strukturbaum um Strukturelement aus ausgewählten Seiten zu erstellen:_
![Bildschirmfoto 2023-06-05 um 12 59 44](https://github.com/kitodo/kitodo-production/assets/19183925/798bca4d-a003-46c7-ae06-367a3705f811)

**Comment by andre-hohmann** [comment]
Vielen Dank für die Antwort - das kann ich nachvollziehen. Mir sind die Unterschiede bisher nicht aufgefallen, aber ich will die Fragen meiner KollegInnen beantworten, die sich in die neuen Abläufe einarbeiten. Vermutlich entstehen einige Fragen, weil intuitiv (Erfahrung aus Kitodo.Production 2) im Strukturbaum strukturiert wird. Unter Umständen sind die Erkenntnisse auch für andere Einrichtungen hilfreich. 

**Fazit**
Somit bestehen folgende Möglichkeiten zur Strukturierung: 
_Vorgänge die keine unstrukturierte Medien enthalten und deren enthaltenen Strukturelemente direkt aufeinanderfolgen_ 
- Strukturierung im Strukturbaum - ohne Einschränkung
- Strukturierung in der Galerie - ohne Einschränkung

_Vorgänge die unstrukturierte Medien enthalten und deren enthaltenen Strukturelemente nicht direkt aufeinanderfolgen_
- Strukturierung in der Galerie - ohne Einschränkung
- Strukturierung im Strukturbaum - Einschränkung: Korrektur der Reihenfolge der Strukturelemente und Medien im Strukturbaum, um Reihenfolge der Objekte zu gewährleisten


**Anmerkungen**
> können wir in der Folge auch nicht sicherstellen, dass aufeinanderfolgende Seiten ohne z.B. ein Kindelement dazwischen ausgewählt wurden

Das Problem besteht jedoch auch bei der Auswahl über ein Strukturelement - siehe Anhang. 

> Das wird aber den hier geschilderten Anforderungen vermutlich nicht gerecht, da es ja darum geht, diese Aktionen komplett - also inkl. Selektion - über den Strukturbaum ausführen zu können).

Wenn nach der Markierung der Seite und Auswahl "Element hinzufügen ..." in dem anschließenden Fenster die zuzuweisenden Seiten manuell angepasst werden könnten, wäre das gegebenenfalls schon ein Fortschritt. Ich stimme jedoch Deinem nachhaltigen Ansatz  zu: 

> To fix this, multiselect for pages should be enabled in the structure trees as well, preferably unifying the data sources for for all three panels in the editor to simplify and streamline the system. 

- #3629

Notwendig wäre aber auch eine Lösung, um gleichzeitig die Objekte (Seiten, ....) prüfen und ansehen zu können. Bisher unterdrückt das starre Fenster alle Funktionen, um zu Blättern, Seiten zu verschieben, um die Seitenzahl zu erkennen, .... Gegebenenfalls könnte anstatt des Fensters neben den Panel "Paginierung" und "Metadaten" auch ein Panel "Strukturdaten" eingeführt werden, das sich öffnet, wenn Strukturelemente hinzugefügt werden. Das sollte jedoch gesondert diskutiert werden. Vergleiche: 
- #4961



**Anhang**
Erstellung eines Strukturelements (Seiten 5-11) mit teilweise bereits strukturierten Seiten (7.8.9). Die Seiten 7-9 werden dem neuen Strukturelement zugewiesen und von dem vorhandenen Strukturelement entfernt. 

![StrukturierenStrukturbaum](https://github.com/kitodo/kitodo-production/assets/14835770/c303e0df-cb8c-4b93-9627-48ea0325828a)
